velox/dwio/parquet/tests/ParquetTpchTestBase.h

```diff
@@ -32,13 +33,14 @@ using namespace facebook::velox::exec;
 using namespace facebook::velox::exec::test;
 using namespace facebook::velox::parquet;
 
-static const int kNumDrivers = 4;
+namespace facebook::velox::exec::test {
+namespace {
 
-class ParquetTpchTest : public testing::Test {
- protected:
+class ParquetTpchTestBase : public testing::Test {
+ public:
   // Setup a DuckDB instance for the entire suite and load TPC-H data with scale
   // factor 0.01.
-  static void SetUpTestSuite() {
+  ParquetTpchTestBase(ParquetReaderType parquetReaderType) {
```

Comment:
We should move the body to `void SetUp() override` so that the derived classes have better control over the initialization.
Add a corresponding `TearDown()` to clean-up the initialization.
See `velox/exec/tests/utils/HiveConnectorTestBase.h` for usage.

Reply:
Thanks for the suggestions, using the older static SetUpTestSuite would require all the member variables to be static actually so I had just moved everything to the constructor and removed that method. I have moved everything to SetUp and TearDown now, please let me know if SetUpTestSuite should be used.